orca-webhook/src/main/groovy/com/netflix/spinnaker/orca/webhook/service/WebhookService.groovy

```diff
@@ -47,30 +58,83 @@ class WebhookService {
-    def restTemplateProvider = restTemplateProviders.find {
-      it.supports(url)
+  ResponseEntity<Object> callWebhook(StageExecution stageExecution) {
+    RestTemplateData restTemplateData = getRestTemplateData('CreateWebhook', stageExecution)
+    if (restTemplateData == null) {
+      throw new SpinnakerException("Unable to determine rest template to call webhook")
     }
+    return  restTemplateData.exchange()
+  }
 
-    URI validatedUri = userConfiguredUrlRestrictions.validateURI(
-      restTemplateProvider.getTargetUrl(url)
-    )
-    HttpHeaders headers = buildHttpHeaders(customHeaders)
-    HttpEntity<Object> payloadEntity = new HttpEntity<>(payload, headers)
-    return restTemplateProvider.getRestTemplate(url).exchange(validatedUri, httpMethod, payloadEntity, Object)
+  ResponseEntity<Object> getWebhookStatus(StageExecution stageExecution) {
+    RestTemplateData restTemplateData = getRestTemplateData('MonitorWebhook', stageExecution)
+    if (restTemplateData == null) {
+      throw new SpinnakerException("Unable to determine rest template to monitor webhook")
+    }
+    return restTemplateData.exchange()
   }
 
-  ResponseEntity<Object> getStatus(String url, Object customHeaders) {
-    def restTemplateProvider = restTemplateProviders.find {
-      it.supports(url)
+  ResponseEntity<Object> cancelWebhook(StageExecution stageExecution) {
+    RestTemplateData restTemplateData = getRestTemplateData("CancelWebhook", stageExecution)
+
+    // Only do cancellation if we can determine the rest template to use.
+    if (restTemplateData == null) {
+      log.warn("Cannot determine rest template to cancel the webhook")
+      return null
+    }
+    WebhookStage.StageData stageData = restTemplateData.stageData
+    try {
+      log.info("Sending best effort webhook cancellation to ${stageData.cancelEndpoint}")
+      ResponseEntity<Object> response = restTemplateData.exchange()
+      log.debug(
+          "Received status code {} from cancel endpoint {} in execution {} in stage {}",
+          response.statusCode,
+          stageData.cancelEndpoint,
+          stageExecution.execution.id,
+          stageExecution.id
+      )
+    } catch (HttpStatusCodeException e) {
+      log.warn("Failed to cancel webhook ${stageData.cancelEndpoint} with statusCode=${e.getStatusCode().value()}", e)
+    } catch (Exception e) {
+      log.warn("Failed to cancel webhook ${stageData.cancelEndpoint}", e)
     }
 
-    URI validatedUri = userConfiguredUrlRestrictions.validateURI(
-      restTemplateProvider.getTargetUrl(url)
-    )
-    HttpHeaders headers = buildHttpHeaders(customHeaders)
-    HttpEntity<Object> httpEntity = new HttpEntity<>(headers)
-    return restTemplateProvider.getRestTemplate(url).exchange(validatedUri, HttpMethod.GET, httpEntity, Object)
+  }
+
+  private RestTemplateData getRestTemplateData(String taskType, StageExecution stageExecution) {
```

Comment:
nit: maybe instead of a "magic" string for `taskType`, use an enum? or the actual task class (but i don't think you have it here) so I think enum would be nicer